### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by ericwbailey**
This PR addresses issues brought up in https://github.com/a11yproject/a11yproject.com/issues/636 and https://github.com/a11yproject/a11yproject.com/issues/631#issuecomment-414445175.

It is a significant overhaul of our `README.md` and `CONTRIBUTING.md` docs. It also builds off of the `STYLE_GUIDE.md` work started by @missmatsuko.

The big-picture goals of this PR are to:

- Update site installation instructions and provide some high-level troubleshooting.
- Provide information about what technology powers the site and how to work with it.
- Move all contribution-related content into a single document.
- Establish authoring standards for the site.
- Outline the expected contribution workflow.

The Files Changed view is tough to parse, so here's links to the updated documents that are easier to read:

- [`README.md`](https://github.com/a11yproject/a11yproject.com/blob/documentation-updates/README.md)
- [`CONTRIBUTING.md`](https://github.com/a11yproject/a11yproject.com/blob/documentation-updates/CONTRIBUTING.md)
- [`CONTENT_STYLE_GUIDE.md`](https://github.com/a11yproject/a11yproject.com/blob/documentation-updates/CONTENT_STYLE_GUIDE.md)

### CODE REVIEW

File: CONTENT_STYLE_GUIDE.md [hunk 1]
```diff
@@ -1,46 +1,321 @@
...
+### Links
+
+- Links should provide context for the content they link to. Avoid using ambiguous terms like "click here".
+- Use Markdown-style links (`[link text](URL)`) instead of HTML.
+- Links should not open in new tabs or windows.
+
+### Images
+
+- Use Markdown-style images (`![alternate description](image url)`) instead of HTML.
+- Provide meaningful alternative (alt) descriptions for images. Alt descriptions should concisely describe the image's content.
```
Comment (scottaohara) [inline CONTENT_STYLE_GUIDE.md]: link to alt text article here https://a11yproject.com/posts/alt-text/

File: CONTENT_STYLE_GUIDE.md [hunk 2]
```diff
@@ -1,46 +1,321 @@
...
+### Code
+
+- Use single backticks to enclose inline code (e.g. The `footer` element typically contains metadata about its section.).
+- Use triple backticks before and after a multi-line block of code.
+- Do not use multi-line blocks of code to create diagrams, flowcharts, or other illustrations.
+
+### Horizontal rules
+
+- Use three hyphens (`---`) to create a horizontal rule.
+- Use horizontal rules to denote thematic breaks in content
```
Comment (scottaohara) [inline CONTENT_STYLE_GUIDE.md]: in "paragraph" content

don't want people to think they cool before headings

File: CONTENT_STYLE_GUIDE.md [hunk 3]
```diff
@@ -1,46 +1,321 @@
...
+- Do not use horizontal rules for decoration.
+
+### Inline HTML
+
+Use HTML only when Markdown cannot accurately describe your content. Use relevant, semantic HTML elements and attributes. Examples of this would be a video embed or a definition list.
```
Comment (scottaohara) [inline CONTENT_STYLE_GUIDE.md]: mention `<kbd>` and that other text-level semantics may not have markdown equivalents

File: CONTENT_STYLE_GUIDE.md [hunk 4]
```diff
@@ -1,46 +1,321 @@
...
+> danah boyd is a technology and social media scholar. She is a Principal Researcher at Microsoft Research, the founder and president of Data & Society Research Institute, and a Visiting Professor at New York University.
 
-In some cases, such as the hashtag (#a11y) or a brand name (A11yProject) are acceptable for greater post exposure and brand loyalty, respectively.
+### Other proper nouns
 
-## Complexity
-- **Short** - Aim for a timed reading length of approximately two minutes.
-- **Focused** - Keep it digestible and to a single topic. Articles that span multiple areas and topics are better broken up.
-- **Accessible** - Strive to use plain language, avoid jargon if possible, and explain complicated concepts.
+These terms are commonly used on the site, or in the accessibility community.
```
Comment (scottaohara) [inline CONTENT_STYLE_GUIDE.md]: add links to terms?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,182 +1,187 @@
...
-## Git and GitHub
-The A11Y Project is an Open Source project. It uses [Git](https://git-scm.com/) and [GitHub](http://github.com/) to handle coordinating maintenance of the site and the publishing of new content and features.
 
-People working on more sophisticated aspects of the site (new features, bug fixes, etc.) are expected to have at least some familiarity with version control, GitHub's key functionality, and using [feature branches to work locally](#set-up-locally).
+## About this site
+
+Following is information about how the site operates. Instructions on how to get the site running locally on your computer can be found in the [`README`](https://github.com/a11yproject/a11yproject.com/blob/gh-pages/README.md).
+
+### Git and GitHub
+The A11Y Project is an [Open Source](https://en.m.wikipedia.org/wiki/Open-source_software) project. It uses [Git](https://git-scm.com/) and [GitHub](http://github.com/) to handle publishing new content and features, as well as coordinating maintenance.
```
Comment (scottaohara) [inline CONTRIBUTING.md]: http://github.com/ => https://github.com/

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,182 +1,187 @@
...
+### Build Process
+This project uses [Jekyll](https://jekyllrb.com/) to convert project files into a website that a browser can read and display. You'll need to make sure you've run all the commands in our [`README`](https://github.com/a11yproject/a11yproject.com#local-development) to get Jekyll working.
+
+Any changes you save will cause the local version of the site to recompile. You can then refresh the browser tab that has your local instance of the site open to preview your edits.
+
+### Important Branches
+
+`gh-pages` is the [production branch](https://www.atlassian.com/git/tutorials/using-branches). This is the live website.
+
+If you have a feature request, we suggest first [filing an Issue](#reporting-issues) to discuss it. Once that feature has been approved you can start coding! Create a new [feature branch](https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow) from `gh-pages`, and name it something that references the feature you'll be working on. For example a feature to increase the site's type size could be: `increase-font-size`.
```
Comment (scottaohara) [inline CONTRIBUTING.md]: i'd expect the "filing an issue" link to go to the new issue screen.  maybe do that and then do a (read: 'reporting issues') where that's the link?

**Comment by ericwbailey** [comment]
I'll keep this open for another week, then assume no news is good news and merge.

File: CONTENT_STYLE_GUIDE.md (hunk 4 above)
Comment (scottaohara) [reply CONTENT_STYLE_GUIDE.md]: We still need to talk about this?

**Pull request merged**